gatsby/src/hooks/useLocationHash.js

```diff
@@ -0,0 +1,15 @@
+import { useMemo } from 'react';
+
+export default function useLocationHash(deps = []) {
+  return useMemo(() => {
+    if (typeof window === 'undefined') {
+      return {};
+    }
+    const hash = {};
+    const params = new URLSearchParams(window.location.hash.slice(1));
+    for (const [key, value] of params) {
+      hash[key] = value;
+    }
+    return hash;
+  }, deps);
+}
```

Comment:
Hook that i didn't end up using because of the way gatsby does SSR.

Reply:
I would normally ask if we should remove this since it's unused, but we can create an issue for that later in order to get these changes to production.